R/profile_emissions.R

```diff
@@ -21,6 +21,7 @@ profile_emissions <- function(companies,
     high_threshold = high_threshold
   ) |>
     add_co2(co2, jitter_amount = option_jitter_amount()) |>
+    best_case_worst_case(best_case_worst_case_emission_profile) |>
```

Comment:
> A functional is a function that takes a function as an input and returns a vector as output. --https://adv-r.hadley.nz/functionals.html

While functionals can be useful, here specifically I think it would be much simpler to use polymorphism. 

A polymorphic  `best_case_worst_case()` would work both with the `tilt_profile` object (so it works here) and also with the `product` dataframe (so it works more generally).

See my previous comment about this and let me know if you'd like me to push a PR off this PR so that you can see what that kind of implementation would look like.

You may want to see this chapter

https://adv-r.hadley.nz/s3.html

Reply:
@maurolepore As suggested by you on slack, I have not used polymorphism in my next commit, but we should use it when best and worst case will be applied on sector profile and transition risk score.